src/org/opensolaris/opengrok/analysis/c/Cxx.lexh

```diff
@@ -0,0 +1,27 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2008, 2016, Oracle and/or its affiliates. All rights reserved.
+ * Portions Copyright (c) 2017, Chris Fraire <cfraire@me.com>.
+ */
+
+Identifier = [a-zA-Z_] [a-zA-Z0-9_]*
+
+Number = (0[xX][0-9a-fA-F]+|[0-9]+\.[0-9]+|[1-9][0-9]*)(([eE][+-]?[0-9]+)?[ufdlUFDL]*)?
```

Comment:
Isn't this a dup ? 
shouldn't we have just one include for both ?
btw. Identifier and Number definitions are most probably shared between lots of languages in exactly the same form as above ...

Reply:
but I don't have a list of languages where this actually really is like that, so for some it might make sense to share identifier and number definition, for others they will need improvement

Reply:
I would argue the standards evolve at their own pace. E.g., C++14 adds support for binary literals — 0b11111100 — but that's not yet handled by this {Number} regex and not necessarily supported either by C.